test/e2e/specs/convert-block-type.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Internal dependencies
+ */
+import {
+	getEditedPostContent,
+	newPost,
+	insertBlock,
+} from '../support/utils';
+
+describe( 'Code block', () => {
+	beforeEach( async () => {
+		await newPost();
+	} );
+
+	it( 'should convert to a preformatted block', async () => {
+		const code = 'print "Hello Dolly!"';
+
+		await insertBlock( 'Code' );
+
+		await page.type( '.editor-block-list__block textarea', code );
+
+		// Verify the content starts as a Code block.
+		const originalPostContent = await getEditedPostContent();
+		const codeRegex = new RegExp( `<!-- wp:code -->\\s*<pre class="wp-block-code"><code>${ code }<\\/code><\\/pre>\\s*<!-- \\/wp:code -->` );
+
+		expect( originalPostContent ).toMatch( codeRegex );
+
+		// Hover over this block to show its toolbar so we can click on the block
+		// conversion button.
+		const insertionPoint = await page.$( '.editor-block-list__block-edit' );
```

Comment:
This could stand to be a helper too, just let me know. 🤷‍♂️

Reply:
Oh yeas, please. We use it in a few places with a different implementation.